packages/regenerator-transform/src/visit.js

```diff
@@ -243,31 +255,20 @@ function getMarkedFunctionId(funPath) {
   return t.clone(markedId);
 }
 
-function renameArguments(funcPath, getArgsId) {
-  let state = {
-    didRenameArguments: false,
-    getArgsId: getArgsId
-  };
-
-  funcPath.traverse(argumentsVisitor, state);
-
-  // If the traversal replaced any arguments references, then we need to
-  // alias the outer function's arguments binding (be it the implicit
-  // arguments object or some other parameter or variable) to the variable
-  // named by argsId.
-  return state.didRenameArguments;
-}
-
-let argumentsVisitor = {
+let argumentsThisVisitor = {
   "FunctionExpression|FunctionDeclaration": function(path) {
     path.skip();
   },
 
   Identifier: function(path, state) {
     if (path.node.name === "arguments" && util.isReference(path)) {
       util.replaceWithOrRemove(path, state.getArgsId());
-      state.didRenameArguments = true;
+      state.usesArguments = true;
     }
+  },
+
+  ThisExpression: function(path, state) {
+    state.usesThis = true;
```

Comment:
I suppose we could abandon the traversal once both `state.usesArguments` and `state.usesThis` have been set to `true`, though I'm not sure that case will be common enough to justify the extra complexity.

Reply:
We can't, because `util.replaceWithOrRemove(path, state.getArgsId());` needs to be run even if `usesArguments` is already `true`.